pallets/settlement/src/lib.rs

```diff
@@ -616,9 +625,10 @@ decl_module! {
             trade_date: Option<T::Moment>,
             value_date: Option<T::Moment>,
             legs: Vec<Leg>,
+            instruction_memo: Option<InstructionMemo>
         ) {
             let did = Identity::<T>::ensure_perms(origin)?;
-            Self::base_add_instruction(did, venue_id, settlement_type, trade_date, value_date, legs)?;
+            Self::base_add_instruction(did, venue_id, settlement_type, trade_date, value_date, legs, instruction_memo)?;
```

Comment:
Don't add the `memo` parameter to the existing extrinsics `add_instruction` and `add_and_affirm_instruction`.  These extrinsics will just pass `None` to `base_add_instruction` for the memo.  Create a new extrinsic `add_instruction_with_memo` to provide the memo.